rlclientlib/utility/context_helper.cc

```diff
@@ -1,65 +1,55 @@
-   * called _multi under the root name space.  {_multi:[{"name1":"val1"},{"name1":"val1"}]}
-   *
-   * \param count   : Return value passed in as a reference.
-   * \param context : String with context json
-   * \param status  : Pointer to api_status object that contains an error code and error description in
-   *                  case of failure
+   * \brief Get the event IDs from the slots entries in the context json string.
+   * 
+   * \param context   : String with context json
+   * \param event_ids : Reference to the mapping from slot index to event ID string, results will be
+   *                    put in here
+   * \param trace     : Pointer to the trace logger
+   * \param status    : Pointer to api_status object that contains an error code and error description in
+   *                    case of failure
    * \return  error_code::success if there are no errors.  If there are errors then the error code is
    *          returned.
    */
-  int get_action_count(size_t& count, const char *context, i_trace* trace, api_status* status) {
+  int get_event_ids(const char* context, std::map<size_t, std::string>& event_ids, i_trace* trace, api_status* status) {
     try {
-      const auto scontext = sutil::to_string_t(std::string(context));
-      auto json_obj = web::json::value::parse(scontext);
-      if ( json_obj.has_array_field(multi) ) {
-        auto const arr = json_obj.at(multi).as_array();
-        count = arr.size();
-        if ( count > 0 )
-          return reinforcement_learning::error_code::success;
-        RETURN_ERROR_LS(trace, status, json_no_actions_found);
+      Document obj;
+      obj.Parse(context);
+
+      if (obj.HasParseError()) {
+        RETURN_ERROR_LS(trace, status, json_parse_error) << "JSON parse error: " << GetParseErrorFunc(obj.GetParseError()) << " (" << obj.GetErrorOffset() << ")";
       }
-      RETURN_ERROR_LS(trace, status, json_no_actions_found);
-    }
-    catch ( const std::exception& e ) {
-      RETURN_ERROR_LS(trace, status, json_parse_error) << e.what();
-    }
-    catch ( ... ) {
-      RETURN_ERROR_LS(trace, status, json_parse_error) << error_code::unknown_s;
-    }
-  }
 
-  int get_event_ids(const char* context, std::map<size_t, std::string>& event_ids, i_trace* trace, api_status* status) {
-    try {
-      const auto scontext = sutil::to_string_t(std::string(context));
-      auto json_obj = web::json::value::parse(scontext);
-      if ( json_obj.has_array_field(slots) ) {
-        auto const arr = json_obj.at(slots).as_array();
-        for (int i = 0; i < arr.size(); i++) {
-          auto current = arr.at(i);
-          if(current.has_string_field(event_id))
-          {
-            auto event_id_string = current.at(event_id).as_string();
+      const Value::ConstMemberIterator itr = obj.FindMember(slots);
+      if (itr != obj.MemberEnd() && itr->value.IsArray()) {
+        const auto arr = itr->value.GetArray();

```

Comment:
shouldn't this be a ref?

Reply:
Probably. Are you referring to `itr` or `arr`? Or possibly both?